crypto/evp/p_lib.c

```diff
@@ -654,6 +654,38 @@ static void EVP_PKEY_free_it(EVP_PKEY *x)
 #endif
 }
 
+static int print_reset_indent(BIO **out, int pop_f_prefix, long saved_indent)
+{
+    BIO_set_indent(*out, saved_indent);
+    if (pop_f_prefix) {
+        BIO *next = BIO_pop(*out);
+
+        BIO_free(*out);
+        *out = next;
+    }
+    return 1;
+}
+
+static int print_set_indent(BIO **out, int *pop_f_prefix, long *saved_indent,
+                            long indent)
+{
+    *pop_f_prefix = 0;
+    *saved_indent = 0;
+    if (indent > 0) {
+        if (BIO_set_indent(*out, indent) <= 0) {
+            if ((*out = BIO_push(BIO_new(BIO_f_prefix()), *out)) == NULL)
+                return 0;
+            *pop_f_prefix = 1;
+        }
+        *saved_indent = BIO_get_indent(*out);
```

Comment:
This saves the indent you just set - which I don't think is what you intended. Doesn't this have to happen first?

Reply:
Ah, right

Reply:
Fixed